tools/bin/find_non_rate_limited_PAT

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+
+# Run away from anything even a little scary
+set -o nounset # -u exit if a variable is not set
+set -o errexit # -f exit for any command failure
+# text color escape codes (please note \033 == \e but OSX doesn't respect the \e)
+blue_text='\033[94m'
+red_text='\033[31m'
+default_text='\033[39m'
+
+# set -x/xtrace' uses a Sony PS4 for more info
+PS4="$blue_text"'${BASH_SOURCE}:${LINENO}:default_text '
+
+# Uncomment for debugging
+# set -o xtrace  # -x display every line before exectution; enables PS4
+
+# @ is the entire CLI arg... ie `this_script arg1 arg2`
+# fun fact, any string with spaces is an array in shell!
+for personal_access_token in $@
+  do
+    echo -e "$blue_text""Curl-ing rate_limit for PAT:  \"$personal_access_token\"""$default_text"
+
+
+    full_response=$(
+      curl  --silent \
+            --header "Authorization: Bearer $personal_access_token" \
+            --request GET https://api.github.com/rate_limit \
+    )
+
+    echo -e "$blue_text""first 4 lines of Curl response:\n$(echo $full_response | head -n4)\n\n""$default_text"
+
+    echo -e "$blue_text""Running jq '.resources.core.remaining' on \$full_response""$default_text"
+
+    rate_limit_remaining=$(echo "$full_response" | jq '.resources.core.remaining' )
+
+    echo -e "$blue_text""Github says this PAT has ""$red_text""$rate_limit_remaining""$blue_text"" left""$default_text"
+
+    # If GH rate limit response > 100
+    if test $rate_limit_remaining -gt 100; then
+      # github actions will NOT pass a string that looks like a secret
+      base64_valid_pat=$(echo "$personal_access_token" | base64)
+      echo -e "$blue_text""Found a good PAT!!""$default_text"
+      # ::set-output name is a github action magic string for output
+      echo "::set-output name=pat::$base64_valid_pat"
```

Comment:
I tried sending this to `dev/null` but GitHub seems to need STDOUT to this steam to read the variable